comp/core/config/config_sync.go

```diff
@@ -0,0 +1,53 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"reflect"
+	"time"
+
+	apiutils "github.com/DataDog/datadog-agent/pkg/api/util"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// syncConfigWithCoreAgent fetches the config from the core agent and updates the local config
+func syncConfigWithCoreAgent(ctx context.Context, config pkgconfigmodel.ReaderWriter, url *url.URL, refreshInterval time.Duration) {
+	ticker := time.NewTicker(refreshInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			data, err := apiutils.DoGetWithContext(ctx, http.DefaultClient, url.String(), apiutils.LeaveConnectionOpen)
+			if err != nil {
+				log.Warnf("Failed to fetch config from core agent: %v", err)
```

Comment:
We might want to log only once every X consecutive failure to avoid flooding the logs. I guess we mainly want to know when we failed to connect and when we succeeded again. WDYT ?

Reply:
> I guess we mainly want to know when we failed to connect and when we succeeded again

Good remark, I'll fix that !

Reply:
Fixed in [`a99667c` (#23211)](https://github.com/DataDog/datadog-agent/pull/23211/commits/a99667c2a90044a4a019384b40882f247b7b9004)